shared/protocol/src/main/java/io/pravega/shared/protocol/netty/WireCommands.java

```diff
@@ -636,7 +636,7 @@ public static WireCommand readFrom(ByteBufInputStream in, int length) throws IOE
             UUID writerId = new UUID(in.readLong(), in.readLong());
             long eventNumber = in.readLong();
             long expectedOffset = in.readLong();
-            Event event = Event.readFrom(in, length - 8 * 5);
+            Event event = Event.readFrom(in, length - Long.BYTES * 4);
```

Comment:
I´m not sure I follow why we reducing the multiplier from 5 to 4.

Reply:
There are 4 Long fields in the header and 1 long field in the footer. After you read 32 bytes from the header you need to read the event, without the 8 byte footer. The following line reads the request id, but only if there's anything left to read.

I agree with you this is odd. Why did this work originally with a multiplier of 5. If I read the code right, the request Id would never be read and would always be -1. This should have been covered in a unit test.

Reply:
If a Pravega Client with the previous version  (<=7) sends a `ConditionalAppend` to Pravega running version (8) then the length passed to read the `Event` would be `length = Long.Bytes * 5` ). 
This would cause the `Event.readFrom(..)` to throw an `InvalidMessageException` . 
Hence the count is reduced to `Long.Bytes * 4`

Reply:
I think what you are saying is making the length the minimum between the two versions, 7 and 8. Instead, as I have pointed above, the right way is to pass the correct length per version and validate according to the version.

Reply:
Thanks for the review comment Flavio. As I understand the version is never passed as part of the WireCommand. (do let know if I misunderstood your comment)

Reply:
What about this:

https://github.com/pravega/pravega/blob/master/shared/protocol/src/main/java/io/pravega/shared/protocol/netty/WireCommands.java#L54

https://github.com/pravega/pravega/blob/master/shared/protocol/src/main/java/io/pravega/shared/protocol/netty/WireCommands.java#L103

Reply:
Yes, this version is passed only as part of the `io.pravega.shared.protocol.netty.WireCommands.Hello` command. This version is not sent for a ConditionalAppend and the version that was sent during `WireCommands.Hello` is not accessible to `ConditionalAppend`